HW2/P3/mandelbrot.pyx

```diff
@@ -15,63 +17,104 @@ cpdef mandelbrot(np.complex64_t [:, :] in_coords,
+                
+                #get the correct portions of the real and imaginary pixels
+                real_c = AVX.make_float8(
+                    real_part[i][j+7],
+                    real_part[i][j+6],
+                    real_part[i][j+5],
+                    real_part[i][j+4],
+                    real_part[i][j+3],
+                    real_part[i][j+2],
+                    real_part[i][j+1],
+                    real_part[i][j],
+                )
+
+                imaginary_c = AVX.make_float8(
+                    imaginary_part[i][j+7],
+                    imaginary_part[i][j+6],
+                    imaginary_part[i][j+5],
+                    imaginary_part[i][j+4],
+                    imaginary_part[i][j+3],
+                    imaginary_part[i][j+2],
+                    imaginary_part[i][j+1],
+                    imaginary_part[i][j],
+                )
+                
+                real_z = AVX.float_to_float8(0)
+                imaginary_z = AVX.float_to_float8(0)
+                new_iter_value = AVX.float_to_float8(0)
+                
+                for iter in range(max_iterations):                    
+                    #calculate the new magnitude
+                    #magnitude = real_z^2 + imaginary_z^2
+                    squared_real = AVX.mul(real_z, real_z)
+                    squared_imaginary = AVX.mul(imaginary_z, imaginary_z)
+                    magnitude = AVX.add(squared_real, squared_imaginary)
+
+                    #check if the magnitude is under the threshold
+                    #we are really checking this for the 8 values, so we calculate the greater than
+                    #less_than_threshold will be -1 where it is less than, and 0 where greater than
+                    less_than_threshold = AVX.less_than(magnitude, four_float)
+
+                    #if everything in less_than_threshold is 0 (all are greater than threshold)
+                    #if none are under the threshold, break (signs converts all 8 floats to a single int)
+                    if AVX.signs(less_than_threshold) == 0:
+                        break
 
-    return np.array(out_view)
+                    #update the real and imaginary portions
+                    #z = z * z + c
+                    #z = (real_z + imaginary_z * i) * (real_z + imaginary_z * i) + (real_c + imaginary_c * i)
+                    #z = real_z^2 + 2 * real_z * imaginary_z * i - imaginary_z^2 + (real_c + imaginary_c * i)
+                    #z = [real_z^2 - imaginary_z^2 + real_c] + [(2 * real_z * imaginary_z + imaginary_c) * i]
+
+                    #real component of z = real_z^2 - imaginary_z^2 + real_c
+                    temp_real = AVX.add(AVX.sub(squared_real, squared_imaginary), real_c)
+                    
+                    #imaginary component of z = 2 * real_z * imaginary_z + imaginary_c
+                    temp_imaginary = AVX.add(AVX.mul(two_float, AVX.mul(real_z, imaginary_z)), imaginary_c)
```

Comment:
Looks great! The operations are very compact and probably very efficient for the compiler.